// react-fiction/lib/components/Rooms.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(_x, _x2, _x3) { var _again = true; _function: while (_again) { var object = _x, property = _x2, receiver = _x3; desc = parent = getter = undefined; _again = false; if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { _x = parent; _x2 = property; _x3 = receiver; _again = true; continue _function; } } else if ('value' in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } } };

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _reactRedux = require('react-redux');

var _ducksStory = require('../ducks/story');

var room = _react.PropTypes.oneOfType([_react.PropTypes.element, _react.PropTypes.object]);
var rooms = _react.PropTypes.oneOfType([room, _react.PropTypes.arrayOf(room)]);

var Rooms = (function (_Component) {
  _inherits(Rooms, _Component);

  function Rooms() {
    _classCallCheck(this, _Rooms);

    _get(Object.getPrototypeOf(_Rooms.prototype), 'constructor', this).apply(this, arguments);
  }

  _createClass(Rooms, [{
    key: 'componentWillMount',
    value: function componentWillMount() {
      var _this = this;

      _react.Children.forEach(this.props.children, function (route) {
        _this.props.addRoom(route.props.id, {
          id: route.props.id,
          title: route.props.title,
          component: route.props.component
        });

        _this.props.updateData(route.props.id, {});

        if (route.props.start) {
          _this.props.updateCurrentRoom(route.props.id);
        }
      });
    }
  }, {
    key: 'render',
    value: function render() {
      return null;
    }
  }], [{
    key: 'propTypes',
    value: {
      children: rooms,
      addRoom: _react.PropTypes.func.isRequired,
      updateData: _react.PropTypes.func.isRequired,
      updateCurrentRoom: _react.PropTypes.func.isRequired
    },
    enumerable: true
  }]);

  var _Rooms = Rooms;
  Rooms = (0, _reactRedux.connect)(null, { addRoom: _ducksStory.addRoom, updateData: _ducksStory.updateData, updateCurrentRoom: _ducksStory.updateCurrentRoom })(Rooms) || Rooms;
  return Rooms;
})(_react.Component);

exports['default'] = Rooms;
module.exports = exports['default'];
